tests/pubsub/testgossipmembership.nim

```diff
@@ -214,39 +237,76 @@ suite "GossipSub Topic Membership Tests":
-
-    # Ensure the topic is correctly initialized in mesh and gossipsub
-    doAssert gossipSub.mesh.contains(topic), "Topic not found in mesh"
-    doAssert gossipSub.gossipsub.contains(topic), "Topic not found in gossipsub"
-
-    # Simulate 6 peers joining the topic
-    subscribeToTopics(gossipSub, @[topic])
-
-    # Assert that 6 peers have joined the mesh
-    doAssert gossipSub.mesh[topic].len == 6, "Expected 6 peers to join the mesh"
-
-    # Define a simple handler for unsubscribing the peers
-    proc dummyHandler(topic: string, data: seq[byte]): Future[void] {.async.} =
-      discard
-
-    # Simulate 3 peers leaving the topic by unsubscribing them
-    var peersToUnsubscribe = gossipSub.mesh[topic].toSeq()[0 .. 2]
+    let
+      numberOfNodes = 6
+      topic = "foobar"
+      nodes = generateNodes(numberOfNodes, gossip = true)
+      nodesFut = await allFinished(nodes.mapIt(it.switch.start()))
+
+    # Subscribe all nodes to the topic
+    await subscribeNodes(nodes)
+    for node in nodes:
+      node.subscribe(topic, voidTopicHandler)
+
+    # Allow time for subscription propagation
+    await sleepAsync(2 * DURATION_TIMEOUT)
+
+    # Ensure each node is subscribed by checking the gossipsub field
+    for i in 0 ..< numberOfNodes:
+      let currentGossip = GossipSub(nodes[i]) # Rename to 'currentGossip'
+      doAssert currentGossip.gossipsub.hasKey(topic),
+        "Node is not subscribed to the topic"
+
+    # Print mesh status before connecting nodes
+    echo "Initial mesh size:"
+    for i in 0 ..< numberOfNodes:
+      let currentGossip = GossipSub(nodes[i]) # Rename to 'currentGossip'
+      echo "Node ", i, " mesh size: ", currentGossip.mesh.getOrDefault(topic).len
+
+    # Connect all nodes to each other and ensure subscription propagation
+    for x in 0 ..< numberOfNodes:
+      for y in 0 ..< numberOfNodes:
+        if x != y:
+          await waitSub(nodes[x], nodes[y], topic)
+
+    # Allow time for mesh stabilization
+    await sleepAsync(2 * DURATION_TIMEOUT)
+
+    # Print mesh status after stabilization
+    echo "Mesh size after connecting:"
+    for i in 0 ..< numberOfNodes:
+      let currentGossip = GossipSub(nodes[i]) # Rename to 'currentGossip'
+      echo "Node ", i, " mesh size: ", currentGossip.mesh.getOrDefault(topic).len
+
+    # Expected number of peers in the mesh
+    let expectedNumberOfPeers = numberOfNodes - 1
+    for i in 0 ..< numberOfNodes:
+      let currentGossip = GossipSub(nodes[i]) # Rename to 'currentGossip'
+      check:
+        currentGossip.gossipsub[topic].len == expectedNumberOfPeers
+        currentGossip.mesh[topic].len == expectedNumberOfPeers
+        currentGossip.fanout.len == 0
+
+    # Simulate unsubscription of 3 peers
+    let firstNodeGossip = GossipSub(nodes[0]) # Initialize for the first node
+    let peersToUnsubscribe = firstNodeGossip.mesh[topic].toSeq()[0 .. 2]
     for peer in peersToUnsubscribe:
+      firstNodeGossip.unsubscribe(topic, voidTopicHandler)
```

Comment:
You are using `peer` as the iteration variable but the unsubscribed node is `firstNodeGossip`. 

The test should fail due to this and it doesn't in the CI/CD. That being said, I ran it locally and I get a failure for it. Can you double check it?